Examples/ScalarField/SimulationParameters.hpp

```diff
@@ -22,25 +22,54 @@ class SimulationParameters : public SimulationParametersBase
     SimulationParameters(GRParmParse &pp) : SimulationParametersBase(pp)
     {
         // read the problem specific params
-        readParams(pp);
+        read_params(pp);
+        check_params();
     }
 
-    void readParams(GRParmParse &pp)
+    void read_params(GRParmParse &pp)
     {
         // Initial scalar field data
         initial_params.center =
             center; // already read in SimulationParametersBase
-        pp.load("G_Newton", G_Newton, 1.0);
-        pp.load("scalar_amplitude", initial_params.amplitude);
-        pp.load("scalar_width", initial_params.width);
-        pp.load("scalar_mass", potential_params.scalar_mass);
+        pp.load("G_Newton", G_Newton,
+                0.0); // for now the example neglects backreaction
+        pp.load("scalar_amplitude", initial_params.amplitude, 0.1);
+        pp.load("scalar_width", initial_params.width, 1.0);
+        pp.load("scalar_mass", potential_params.scalar_mass, 0.1);
 
         // Initial Kerr data
         pp.load("kerr_mass", kerr_params.mass, 1.0);
         pp.load("kerr_spin", kerr_params.spin, 0.0);
         pp.load("kerr_center", kerr_params.center, center);
     }
 
+    void check_params()
+    {
+        warn_parameter("scalar_mass", potential_params.scalar_mass,
+                       potential_params.scalar_mass <
+                           0.2 / coarsest_dx / dt_multiplier,
+                       "oscillations of scalar field do not appear to be "
+                       "resolved on coarsest level");
+        warn_parameter("scalar_width", initial_params.width,
+                       initial_params.width < 0.5 * L,
+                       "is greater than half the domain size");
```

Comment:
Perhaps this should take into account non-cubic grids and reflective BCs (e.g. in the same way as for the extraction radii)? We could make `reflective_box_lo` and `reflective_box_hi` a member of `ChomboParameters` (and calculate it there) to save having to recalculate this again.

Reply:
We could make them `protected` members so they're not accessible elsewhere.

Reply:
I think it is not really worth it for this particular case, as this is just a very rough sense check - also the data will still be valid with a bigger value, it will just not be a very good simulation. 

But I'd be happy to add it if you think it might be useful for other cases. I guess it might.

Reply:
Ok, I guess for this case it's just a warning so I'm happy to leave it. I can make the change I suggested although I think I'll rename `box` to `domain` as I think the former is a bit ambiguous.